src/battle_ai_switch_items.c

```diff
@@ -1851,13 +1916,26 @@ static u32 GetBestMonIntegrated(struct Pokemon *party, int firstId, int lastId,
                     else
                     {
                         // If AI mon can't be 2HKO'd
-                        if (hitsToKO > hitsToKOThreshold + 1)
+                        if (hitsToKOAI > hitsToKOAIThreshold + 1)
                         {
                             // We have a slow threaten
                             slowThreatenId = i;
                         }
                     }
                 }
+
+                // If mon can trap
+                if (CanAbilityTrapOpponent(AI_DATA->switchinCandidate.battleMon.ability, opposingBattler))
+                {
+                    hitsToKOPlayer = GetNoOfHitsToKOBattlerDmg(damageDealt, opposingBattler);
+                    // Good grief clean up the 1v1 check and outspeed check
```

Comment:
The comment should probably be removed.

Reply:
LOL yeah good catch that was a note-to-self during cleanup